src/v/cloud_storage/remote_partition.cc

```diff
@@ -752,7 +634,8 @@ remote_partition::timequery(storage::timequery_config cfg) {
 
     // Synthesize a log_reader_config from our timequery_config
     storage::log_reader_config config(
-      kafka::offset_cast(start_offset),
+      // TODO: fix
+      start_offset,
```

Comment:
Ah, so this is why TimeQueryTest is failing.

Reply:
fixed that